### ISSUE TITLE
Issue with opTomo operator in 2D fanflat projection geometry

### ISSUE THREAD

**Issue by Joaquim-Sanctorum** [astra-toolbox/astra-toolbox#283]
In a 2D fanflat projection geometry, multiplication of a sinogram by the transpose of the opTomo operator yields a matrix of zeros instead of a non-filtered backprojection. 

I used following code in Matlab 2019b (after adding the ASTRA toolbox and the Spot opTomo package to the path) using the 1.9.0.dev11 version of the ASTRA toolbox on a Windows 10 operating system:

```Matlab
% --------------------------------------------------------------------------------------------------------------------------
% phantom
im = phantom(256,256);
figure; imshow(im,[])

% geometries
proj_geom = astra_geom_2vec(astra_create_proj_geom('fanflat',1,400, linspace2(0,2*pi,360),6000,300));
vol_geom  = astra_create_vol_geom(256,256);

% Create the Spot operator for ASTRA using the GPU
A = opTomo('cuda', proj_geom, vol_geom);

% forward projection
p = A*im(:); 
figure; imshow(reshape(p, A.proj_size), []);

% backprojection
BP = A'*p;
figure; imshow(reshape(BP,256,256), []);  % BP = all zeros???
% ------------------------------------------------------------------------------------------------------------------------
```
While the forward projector A works just fine, multiplying p by A' results in a matrix containing only zeros, while this should be a non-filtered backprojection. This also happens in regular fanflat geometry, and not only the vector geometry as used here.
Some remarks:  
- in a 3D cone example, this does not occur
- replacing 'cuda' by 'line_fanflat' in the creation of A fixes the problem, but this is much slower
- A colleague using the ASTRA 1.9.9.dev1 version in Python (on a Linux operating system) did not encounter this issue, so will this version be available for Matlab too?

**Comment by meetaig** [comment]
The colleague here :)
I tested with 1.9.9dev, but also on MATLAB, with the exact code above. This does not produce any errors. We also tried activating the ASTRA logging using astra_mex_log('enable', 'all'), but it didn't print anything.

**Comment by wjp** [comment]
By "This does not produce any errors.", do you mean that it works properly in 1.9.9.dev in Matlab? Or that it fails silently?

**Comment by meetaig** [comment]
I mean that it produces a correct (blurry) backtransform, as one would expect. There are no fails even in the code Joaquim showed. The result is just empty

**Comment by Joaquim-Sanctorum** [comment]
Thank you both for your replies.
Instead of using the Spot opTomo operator, I have used the 'BP_CUDA' algorithm to obtain a backprojection (see following code) in a fanflat 2D geometry:

```Matlab
        cfg = astra_struct('BP_CUDA');
        cfg.ProjectionDataId =      projectionsId;
        cfg.ReconstructionDataId =  reconstructionId;
        astra_mex('set_gpu_params', options.numberGPU);
        algId = astra_mex_algorithm('create', cfg);
        astra_mex_algorithm('run', algId);

        BP = single(astra_mex_data2d('get', reconstructionId));
```
Still, the BP matrix contains all zeros. So the issue is not necessarily related to the Spot opTomo operator, but to the backprojector itself. Maybe this comment is redundant, but I'm trying to pinpoint the source of the issue :)

**Comment by wjp** [comment]
Do I understand your messages correctly that everything works fine with 1.9.9.dev already, and so only updated packages are needed?

**Comment by Joaquim-Sanctorum** [comment]
As far as we were able to deduce, the backprojection indeed worked in the 1.9.9.dev1 version. However, it seems that I cannot download that version from the toolbox page. So an updated version would be really helpful!

**Comment by meetaig** [comment]
Yes, just an updated version for Windows (pre-compiled binaries) would be needed, I assume. 
Related: https://github.com/astra-toolbox/astra-toolbox/issues/267#issue-779102493
